Lecture_2_Collections_homework/hw1.py

```diff
@@ -29,15 +29,25 @@
 bar@example.com 729.83 accountName 2021-01-02 validate_line
 """
 from typing import Callable, Iterable
+import re
 
 
 def validate_line(line: str) -> bool:
-    ...
+    return len(line.split()) == 5
 
 
 def validate_date(date: str) -> bool:
-    ...
+    regexp = re.compile(r'.*\d{4}-\d{2}-\d{2}$')
+    return bool(regexp.match(date))
```

Comment:
```suggestion
    return re.match(r'.*\d{4}-\d{2}-\d{2}$', date)
```
тут можно не использовать compile (все равно ведь будет компайлиться каждый раз при вызове функции; смысл compile в том чтобы один раз скомпилить и потом много раз применять). Ну и можно сразу возвращать, не кастить в bool Метод match вернет либо матч объект, либо None - и в соответствующем if этот None превратится в False, а любой не пустой объект в True https://docs.python.org/3/library/stdtypes.html#truth-value-testing

Reply:
Fixed